lib/Controller/WopiController.php

```diff
@@ -485,7 +485,13 @@ public function putFile($fileId,
 
 			$content = fopen('php://input', 'rb');
 
+			$freespace = $file->getParent()->getFreeSpace();
+			$contentLength = $this->request->getHeader('CONTENT_LENGTH');
+
 			try {
+				if ($contentLength > $freespace) {
```

Comment:
```suggestion
				if ($freespace >= 0 && $contentLength > $freespace) {
```

Otherwise if there is no quota for various reasons we would always fail as those are indicated by negative numbers https://github.com/nextcloud/server/blob/e62e9e3dbf1a2573554b1a9eabbf5b59b652dae6/lib/public/Files/FileInfo.php#L53-L63